### ISSUE TITLE
IncompatibleClassChangeError: ScenarioStateField$1 does not implement the requested interface java.util.function.Function

### ISSUE THREAD

**Issue by TaylorSMarks** [TNG/JGiven#1380]
I'm using JGiven 1.2.5, JDK 17, JUnit 5, Surefire 3.1.2, and Maven 3.9.0.

No matter what I try to do, I'm always getting this exception stack trace when I try to run tests involving JGiven:

~~~
java.lang.IncompatibleClassChangeError: Class com.tngtech.jgiven.impl.inject.ScenarioStateField$1 does not implement the requested interface java.util.function.Function
    at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
    at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
    at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
    at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
    at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
    at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
    at com.tngtech.jgiven.impl.inject.ValueInjector.getScenarioFields(ValueInjector.java:73)
    at com.tngtech.jgiven.impl.inject.ValueInjector.validateFields(ValueInjector.java:45)
    at com.tngtech.jgiven.impl.inject.ValueInjector.updateValues(ValueInjector.java:102)
    at com.tngtech.jgiven.impl.ScenarioExecutor.updateScenarioState(ScenarioExecutor.java:239)
    at com.tngtech.jgiven.impl.ScenarioExecutor.access$500(ScenarioExecutor.java:41)
    at com.tngtech.jgiven.impl.ScenarioExecutor$StageTransitionHandlerImpl.enterStage(ScenarioExecutor.java:172)
    at com.tngtech.jgiven.impl.intercept.StepInterceptorImpl.intercept(StepInterceptorImpl.java:80)
    at com.tngtech.jgiven.impl.intercept.ByteBuddyMethodInterceptor.interceptSuper(ByteBuddyMethodInterceptor.java:37)
    at com.company.gateway.notification.bdd.scenarios.notification.steps.GivenNotification$ByteBuddy$FF4QBf1B.anXMLNotificationOfType(Unknown Source)
~~~

Any idea what's going on?

**Comment by l-1squared** [comment]
Hi, sorry for missing your post for so long. I happen to know this error, and I am pretty convinced I fixed it on the main branch... If I recall correctly, the method in question uses google function and I guess that confuses the JVM sometimes..

I will provide a fix within this week. I was planning on doing so anyway, but I was hoping to get at least the documentation for the Asciidoc reporter in there. Alas...

**Comment by l-1squared** [comment]
should be fixed by update 1.3.0
